### ISSUE TITLE
Web (react-native-web) will not support ViewPropTypes API it is upcoming release

### ISSUE THREAD

**Issue by vladp** [wix/react-native-calendars#1033]
## Description

React-native-calendars is a very portable package. Some folks, like us, have been using it on web browsers with React-native-web.

With React-native-web release 0.12 and above,  RNW will remove ViewPropTypes export.

This API, unfortunately, is being used by this package, therefore, unless changes are made this will break the web-platform usage.

(tested with most recent release of react-native-calendars 1.221.0  and the upcoming release candidate for RNW -- 0.12.0-rc.1)

**Comment by jpaas** [comment]
Just hit this problem on my upgrade to react-native-web 0.12

**Comment by nathsimpson** [comment]
I have a PR for this, but it's still waiting for review :( https://github.com/wix/react-native-calendars/pull/1089

**Comment by vladp** [comment]
@Inbal-Tish  is that possible to incorporate @nathsimpson PR in?
Or are there wider context issues that prevent the PR from going in?

thx

**Comment by vladp** [comment]
it seems that it was merged yesterday, but not part of the npm release yet

**Comment by vladp** [comment]
@nathsimpson 
The fix is not working for us.
Still getting

```
Creating an optimized production build...
Failed to compile.

./node_modules/react-native-calendars/src/calendar/index.js
Attempted import error: 'ViewPropTypes' is not exported from 'react-native' (imported as 'ReactNative').
```

If I am understanding this correctly, your update checks if ViewPropTypes is defined, and if it is not defined, it aliases to something else.

However webpack (we are using CRA toolchain with config overrides),
seems to 'detect' that ViewPropTypes are still being imported
may be by this line calendars/index.js

`const {View, ViewPropTypes} = ReactNative;
`
Because, if I remove ViewPropTypes from the above line, in this file and in agenda/idnex.js -- the build goes through.

Are you somehow able to tell webpack to ignore the non-existing import?

thank you in advance for any hints

**Comment by nandorojo** [comment]
This is the solution:

```js
// agenda/index.js
import {Text, View, Dimensions, Animated, Platform}  from 'react-native';

...

const viewPropTypes =
  typeof document !== 'undefined' || Platform.OS === 'web'
    ? PropTypes.shape({style: PropTypes.object})
    : require('react-native').ViewPropTypes || View.propTypes;
```

We'd do the same in `calendar/index.js`. This way, we inline require `ViewPropTypes`, only on platforms where it still exists.

@nathsimpson I believe #1089 breaks react-native-web tree shaking by importing the whole package.

I could submit a PR with a fix. In the meantime, I might rely on `patch-package`.

**Comment by nathsimpson** [comment]
That feeling when you're looking up this issue, and people are discussing a PR you made 😅 
I've run into this issue again after updating React Native Web to v0.13.7, while updating Expo and RN to 39 and 0.63.3. Even on latest version of React Native Calendars. Looks like PR #1313 will fix the issue 👍

**Comment by nathsimpson** [comment]
Apologies @vladp for not seeing your comments

**Comment by nandorojo** [comment]
I left some comments on #1313, which should help solve the outstanding issues there. It would be awesome to have it merged. In the meantime, I'll probably have to make the changes on my local branch.

Thanks to everyone helping on this!

**Comment by gstcyr** [comment]
Here's my patch-web.sh script that i keep in a `scripts/` directory. In package.json, just add `"postinstall": "scripts/patch-web.sh"` to the `scripts` section. 

```shell
# needed by some libraries for react-native-web 12+ which no longer exports ViewPropTypes

if test -f "node_modules/react-native-web/dist/patched"; then
  echo "react-native-web already patched!"
else
  echo "Patching react-native-web propTypes"
  echo '\n'"export const ViewPropTypes = { style: ()=>{} };" >> node_modules/react-native-web/dist/index.js

  mkdir -p node_modules/react-native-web/dist/exports/ViewPropTypes
  echo "" >> node_modules/react-native-web/dist/exports/ViewPropTypes/index.js

  # patch Text default propTypes export:
  echo '\n'"Text.propTypes = {
    style: ()=>{}
  }" >> node_modules/react-native-web/dist/exports/Text/index.js

  #patch Image default propTypes export:
  echo '\n'"Image.propTypes = {
    style: ()=>{}
  }" >> node_modules/react-native-web/dist/exports/Image/index.js

  echo "patched" >> node_modules/react-native-web/dist/patched
fi
```

**Comment by bananer** [comment]
Maybe we can just stop using ViewPropTypes? I've created a PR #1424

**Comment by mistkafka** [comment]
@bananer 

I choose remove all props-type, and it's work for me.

**Comment by vladp** [comment]
it seems that there are no upstream fixes in this repo that will resolve this issue.
Not sure this issue, therefore needs to be closed as non-resolvable, or keep it open

**Comment by bananer** [comment]
Since #1424 was merged, react-native-calendards 1.1259.0 works for me with react-native-web 0.16.1

@vladp What are you still missing?
